assets/js/ui-theme-support.js

```diff
@@ -0,0 +1,169 @@
+		value: null,
+	},
+	/**
+	 * Set the background color used for overlay elements to dark/light.
+	 */
+	'--jm-ui-background-color': {
+		init: ( { darkMode, backgroundColor } ) => {
+			if ( backgroundColor ) {
+				// Check if the page background color is dark enough for dark mode's white text, and use a slightly lighter version of it.
+				if ( darkMode && backgroundColor.l < 0.5 ) {
+					return `color-mix(in srgb, #fff, ${ backgroundColor.css } 95%)`;
+				}
+
+				if ( ! darkMode ) {
+					return `color-mix(in srgb, #fff, ${ backgroundColor.css } 50%)`;
+				}
+			}
+			return darkMode ? '#111' : '#fff';
+		},
+		value: null,
+	},
+	/**
+	 * Use the text color for the UI text color if it has enough contrast with the background.
+	 */
+	'--jm-ui-text-color': {
+		init: ( { darkMode, textColor } ) => {
+			if ( textColor ) {
+				// Use theme text color if it has enough contrast with the background.
+				if ( ( darkMode && textColor.l < 0.25 ) || ( ! darkMode && textColor.l > 0.25 ) ) {
+					return textColor.css;
+				}
+			}
+			return darkMode ? '#fff' : '#000';
+		},
+		value: null,
+	},
+	/**
+	 * Create a darker or lighter blend of the accent color, based on its brightness.
+	 */
+	'--jm-ui-accent-alt-color': {
+		init: ( { linkColor, backgroundColor } ) => {
+			const lighten = linkColor.l < 0.2;
+			const darkBase = backgroundColor?.l < 0.4 ? backgroundColor.css : '#000';
+			const base = lighten ? '#fff' : darkBase;
+			const amount = lighten ? '65%' : '40%';
+
+			return `color-mix(in srgb, ${ base }, var(--jm-ui-accent-color) ${ amount })`;
+		},
+		value: null,
+	},
+	/**
+	 * Use the background color for the button text color if it has enough contrast.
+	 */
+	'--jm-ui-accent-color-contrast': {
+		init: ( { linkColor, backgroundColor } ) => {
+			if ( Math.abs( backgroundColor?.l - linkColor?.l ) > 0.5 ) {
+				return backgroundColor.css;
+			}
+			return '#fff';
+		},
+		value: null,
+	},
+	/**
+	 * Adjust the red danger/error color to be legible on dark backgrounds.
+	 */
+	'--jm-ui-danger-color': {
+		init: ( { backgroundColor } ) => {
+			if ( backgroundColor?.l < 0.5 ) {
+				return `var(--jm-ui-danger-color-dark-mode)`;
+			}
+		},
+		value: null,
+		default: '#cc1818',
+	},
+	/**
+	 * Try to use the font family (typically sans-serif) used by the theme for menus and other elements, rather than the serif font typically used for post content.
+	 */
+	'--jm-ui-font-family': {
+		init: ( { bodyFontFamily } ) => bodyFontFamily,
+		value: null,
+	},
+};
+
+/**
+ * Detect and compute style variables based on the site theme.
+ * Skips any variable that's been explicitly set already.
+ */
+function applyThemeStyles() {
+	const style = getComputedStyle( document.documentElement );
+	for ( const [ key, color ] of Object.entries( uiVariables ) ) {
+		let value = style.getPropertyValue( key );
+
+		const hasCustomValue = color.default ? color.default !== value : !! value;
+
+		if ( ! hasCustomValue ) {
```

Comment:
So IIUC this code should detect if the user has set any of the CSS variables and use these values instead right? So I am trying to specify these values but they get ignored. What I did was adding this in the customizer:
```
:root {
	--jm-ui-background-color: red;
	--jm-ui-accent-color: red;
	--jm-ui-text-color: red;
	--jm-ui-accent-alt-color: red;
}
```
But the background has still the same color.

Probably this gets overriden by `	--jm-ui-background-color: initial;` in neutral.scss as when I remove this it works.

Also for `--jm-ui-accent-color-contrast` specifically, I think that the calculations from ThemeStyles are always ignored as we don't have a default value in Color and we set a default value of #ffffff in neutral.scss. Was this intentional?

Reply:
Yup that's how it's supposed to work, if the colors are defined manually then this shouldn't override them. I'll take a look.